libtiledbsoma/src/reindexer/reindexer.cc

```diff
@@ -51,14 +51,6 @@ void IntIndexer::map_locations(const int64_t* keys, int size, int threads) {
     if (size == 0) {
```

Comment:
several concerns here with param checking:
1. do you want map_locations to create a thread pool in the case where `threads == 1`?  It looks like it will never be used.
2. Why is `threads` a signed value?  ThreadPool uses `size_t`.  If there is meaning to negative values, don't you need to bounds check the arg?
3. What happens if `size` is <= 0?  Does this need a check?  Should size be unsigned?

Reply:
@beroy please let's do a follow-up. I shouldn't have approved this; @bkmartinjr 's eyes are sharper than mine. :)

Reply:
Follow up PR: https://github.com/single-cell-data/TileDB-SOMA/pull/2098